extensions/vertx-http/deployment/src/main/java/io/quarkus/devui/deployment/menu/AuthProcessor.java

```diff
@@ -0,0 +1,274 @@
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.deployment.IsDevelopment;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ConfigDescriptionBuildItem;
+import io.quarkus.deployment.builditem.DevServicesLauncherConfigResultBuildItem;
+import io.quarkus.dev.console.DevConsoleManager;
+import io.quarkus.devui.deployment.InternalPageBuildItem;
+import io.quarkus.devui.runtime.config.*;
+import io.quarkus.devui.spi.JsonRPCProvidersBuildItem;
+import io.quarkus.devui.spi.page.Page;
+import io.smallrye.config.SmallRyeConfig;
+
+/**
+ * This creates Extensions Page
+ */
+public class AuthProcessor {
+
+    private static final String NAME_SPACE = "devui-auth";
+    private static List<ConfigDescriptionBuildItem> configDescriptionBuildItems;
+    private static Optional<DevServicesLauncherConfigResultBuildItem> devServicesLauncherConfig;
+    private static AuthDevUIRecorder recorder;
+    private static SmallRyeConfig current;
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    InternalPageBuildItem createAuthPages() {
+
+        InternalPageBuildItem configurationPages = new InternalPageBuildItem("Auth", 21);
+
+        configurationPages.addPage(Page.webComponentPageBuilder()
+                .namespace(NAME_SPACE)
+                .title("Permissions")
+                .icon("font-awesome-solid:shield")
+                .componentLink("qwc-auth.js"));
+
+        return configurationPages;
+    }
+
+    private Optional<ConfigDescriptionBuildItem> findByName(String propertyKey,
+            List<ConfigDescriptionBuildItem> configDescriptionBuildItems) {
+        Pattern pattern = PermissionDescription.getPattern(propertyKey);
+
+        for (ConfigDescriptionBuildItem configDescriptionBuildItem : configDescriptionBuildItems) {
+            boolean matches = configDescriptionBuildItem.getPropertyName().matches(pattern.pattern());
+            if (matches) {
+                return Optional.of(configDescriptionBuildItem);
+            }
+
+        }
+        return Optional.empty();
+    }
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    @Record(ExecutionTime.RUNTIME_INIT)
+    void registerConfigs(List<ConfigDescriptionBuildItem> configDescriptionBuildItems,
+            Optional<DevServicesLauncherConfigResultBuildItem> devServicesLauncherConfig,
+            AuthDevUIRecorder recorder) {
+
+        AuthProcessor.configDescriptionBuildItems = configDescriptionBuildItems;
+        AuthProcessor.devServicesLauncherConfig = devServicesLauncherConfig;
+        AuthProcessor.recorder = recorder;
+        AuthProcessor.current = (SmallRyeConfig) ConfigProvider.getConfig();
```

Comment:
Maybe this hack could work as we are talking DEV mode, but if it works it's IMO bug. 
You can only work with runtime configuration when `io.quarkus.deployment.builditem.RuntimeConfigSetupCompleteBuildItem` is produced and you should do this inside recorder because not all config sources will be available during build time. I remember there is missing feature in SR Config that classic config sources in META-INF are also loaded at build time, but Quarkus also provides runtime-only config sources defined by `io.quarkus.deployment.builditem.RunTimeConfigBuilderBuildItem`.